packages/slice-machine/src/modules/selectedSlice/sagas.ts

```diff
@@ -0,0 +1,328 @@
+import {
+  call,
+  fork,
+  takeLatest,
+  put,
+  SagaReturnType,
+} from "redux-saga/effects";
+import { getType } from "typesafe-actions";
+import { withLoader } from "../loading";
+import { LoadingKeysEnum } from "../loading/types";
+import {
+  generateSliceScreenshotCreator,
+  generateSliceCustomScreenshotCreator,
+  saveSliceCreator,
+  pushSliceCreator,
+} from "./actions";
+import {
+  generateSliceScreenshotAxios,
+  generateSliceCustomScreenshotAxios,
+  saveSliceAxios,
+  pushSliceAxios,
+  renameSlice,
+  getState,
+} from "@src/apiClient";
+import { openToasterCreator, ToasterType } from "@src/modules/toaster";
+import { renameSliceCreator } from "../slices";
+import { modalCloseCreator } from "../modal";
+import { ModalKeysEnum } from "../modal/types";
+import { push } from "connected-next-router";
+
+export function* generateSliceScreenshotSaga({
+  payload,
+}: ReturnType<typeof generateSliceScreenshotCreator.request>) {
+  const { component, setData } = payload;
+  try {
+    setData({
+      loading: true,
+      done: false,
+      error: null,
+      status: null,
+      imageLoading: true,
+    });
+    const response = (yield call(
+      generateSliceScreenshotAxios,
+      component.model.name,
+      component.from
+    )) as SagaReturnType<typeof generateSliceScreenshotAxios>;
+    if (response.status > 209) {
+      return setData({
+        loading: false,
+        done: true,
+        error: response.data.err,
+        status: response.status,
+        message: response.data.reason,
+        imageLoading: false,
+      });
+    }
+    setData({
+      loading: false,
+      done: true,
+      error: null,
+      warning: !!response.data.warning,
+      status: response.status,
+      message: response.data.warning || "Screenshots were saved to FileSystem",
+      imageLoading: false,
+    });
+    yield put(
+      generateSliceScreenshotCreator.success({
+        screenshots: response.data.screenshots,
+        component,
+      })
+    );
+  } catch (e) {
+    yield put(
+      openToasterCreator({
+        message: "Internal Error: Screenshot not saved",
+        type: ToasterType.ERROR,
+      })
+    );
+  }
+}
+
+export function* generateSliceCustomScreenshotSaga({
+  payload,
+}: ReturnType<typeof generateSliceCustomScreenshotCreator.request>) {
+  const { variationId, component, setData, file } = payload;
```

Comment:
i'm super surprised that you can pass functions into an action's Payload, thought actions were like raw data. Nice 👍